### ISSUE TITLE
Incompatible shapes error when running object detection train.py locally

### ISSUE THREAD

**Issue by SaleheErfanianEbadi** [tensorflow/models#1887]
I'm trying to run the train.py in object detection locally with my own dataset that I made to look like PASCAL VOC 2012. I can generate the tfrecord files. I get the following output followed by some errors:

![image](https://user-images.githubusercontent.com/23528348/27966455-361b60e8-6337-11e7-8b4e-aaf0f40ae03c.png)

followed by
    
    InvalidArgumentError (see above for traceback): Incompatible shapes: [1,62,4] vs. [1,64,4]
	 [[Node: gradients/Loss/BoxClassifierLoss/Loss/sub_grad/BroadcastGradientArgs = BroadcastGradientArgs[T=DT_INT32, _device="/job:localhost/replica:0/task:0/gpu:0"](gradients/Loss/BoxClassifierLoss/Loss/sub_grad/Shape, gradients/Loss/BoxClassifierLoss/Loss/sub_grad/Shape_1)]]
	 [[Node: gradients/FirstStageFeatureExtractor/InceptionResnetV2/InceptionResnetV2/Repeat_1/block17_14/Branch_0/Conv2d_1x1/BatchNorm/batchnorm/sub_grad/tuple/control_dependency/_6423 = _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/cpu:0", send_device="/job:localhost/replica:0/task:0/gpu:0", send_device_incarnation=1, tensor_name="edge_35146_gradients/FirstStageFeatureExtractor/InceptionResnetV2/InceptionResnetV2/Repeat_1/block17_14/Branch_0/Conv2d_1x1/BatchNorm/batchnorm/sub_grad/tuple/control_dependency", tensor_type=DT_FLOAT, _device="/job:localhost/replica:0/task:0/cpu:0"]()]]

Any ideas why this happens?

**Comment by SaleheErfanianEbadi** [comment]
Labels need to start from 1 and not 0. This solved the issue.

**Comment by imucsliurui** [comment]
why?
